### ISSUE TITLE
2 nuand blade rf

### ISSUE THREAD

**Issue by jk28jan1990** [JoseAmador95/UoS_Beamforming#3]
Hi , 
Firstly I would congratulate you for this fantastic work. 
Secondly , I would like to know if i need extra hardware in case i use 2 nuand blade rf , 1 for transmiter and other for reciever. 
Thnak you.

**Comment by JoseAmador95** [comment]
Hi! Can you tell me a bit more about what you aim to achieve?

**Comment by jk28jan1990** [comment]
Dear,
Thank you for your reply.
In your setup you have used one BladeRf and connected 2 transmitting antennas and 2 receiving antennas with SMAs. I trying to create a similar system like yours but with 2 Nuand BladeRFs. 2 transmitting antennas will be on one BladeRF and 2 receiving antennas on another BladeRF. So that overall system will be wireless. I will be grateful if you could guide me a little in terms of extra hardware requirement. I am curious how will 2 Nuand BladeRFs will communicate. And later measure the DOA at different positions.





[An Ink Drawing]

  Regards,
  Jaspreet Kaur
PhD Scholar
James Watt School of Engineering
College of Science and Engineering
University of Glasgow
Glasgow G12 8QQ

E: ***@***.******@***.***>
      ***@***.******@***.***>
W:www.gla.ac.uk/schools/engineering/<http://www.gla.ac.uk/schools/engineering/>
***@***.***
www.gla.ac.uk/<http://www.gla.ac.uk/>

**Comment by JoseAmador95** [comment]
I think you will not need additional hardware. As far as I understood, you are trying to replicate my work (2x2 Beamformer), but by using two SDRs. Well, you would be extending it, as both devices could track the other.

I was able to do that with a student with access to the hardware, the challenge was to tweak the carrier synchronisation to handle the difference in the local oscillators. Honestly, we did it by trial and error. I would suggest driving each SDR with a different computer, although we did not have apparent issues when using a single PC for both SDRs, but MATLAB adds a lot of overhead and could cause frame drops.

**Comment by jk28jan1990** [comment]
Dear,
Thank you so much for the details. Have you posted the work of your student?
if yes,  could you share the link or report? because my aim is just collecting the DOA at different positions as a real time data and use it in another code for theoretically calculated ones?
Thank you.


Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by JoseAmador95** [comment]
I do not have access to her work. I'm not a professor, she was a student who landed on my repository (like you, I suppose haha). But let me know if I can help you set things up!

**Comment by jk28jan1990** [comment]
Many thanks. Let me try what you have suggested. I will update you with results or error, I will be getting.
Thank you again .


Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by jk28jan1990** [comment]
Dear,
I have tried to introduce 2 SDRs, I connected both with Txer and it seems working. But I am not sure. Would you check it, please? Also, can you tell me how one can see the change in DOA as shown in your video?
***@***.***


Regards,
  Jaspreet Kaur
PhD Scholar
James Watt School of Engineering
College of Science and Engineering
University of Glasgow
Glasgow G12 8QQ

E: ***@***.******@***.***>
      ***@***.******@***.***>
W:www.gla.ac.uk/schools/engineering/<http://www.gla.ac.uk/schools/engineering/>
***@***.***
www.gla.ac.uk/<http://www.gla.ac.uk/>

**Comment by JoseAmador95** [comment]
Of course! Let me know how do you want to show your progress (call, video...)

I can't remember the block's name, but I was plotting the _Spatial Spectrum_. There should be a block with a similar name.

**Comment by JoseAmador95** [comment]
I may have used [this block](https://www.mathworks.com/help/phased/ref/musicspectrum.html). There are other blocks for the different beamformers.

**Comment by jk28jan1990** [comment]
Dear,
I am interested in angle values also power and time if possible, when I will place the Rxer side at different position.
If call or video is possible,  that's ok. But main goal is find the change in angle when Rxer side will place from point a to point b. The way you have shown in GitHub page video. One side is moving and curve is moving as well.


Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by JoseAmador95** [comment]
Here is my contact, the link expires tomorrow.
https://sam.nl.tab.digital/s/XEfwZXL4opHDe94
